includes/codegen/controls/AbstractControl_CodeGenerator.class.php

```diff
@@ -0,0 +1,79 @@
+<?php
```

Comment:
Now that I renamed the classes to have the "Base" suffix, we already have `ControlBase_CodeGenerator` which contains the actual generic implementation. We can make that class the root of this hierarchy, but I kind a like having the (almost) pure abstract class which kinda plays the role of an interface and documents which methods are expected to be implemented.
If you'd like, I could actually make that into a real interface (and move the little code that's there into `ControlBase_CodeGenerator`).